maintainer/span/span_controller.go

```diff
@@ -369,6 +369,7 @@ func (c *Controller) MarkSpanReplicating(span *replica.SpanReplication) {
 func (c *Controller) BindSpanToNode(old, new node.ID, span *replica.SpanReplication) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
+	span.SetNodeID(new)
```

Comment:
This has been set in the following line of the file https://github.com/pingcap/ticdc/blob/1946e51dd173c89363e04b37c73a4c6b5876e0f4/pkg/scheduler/replica/replication_group.go#L143 which is called in the function `BindReplicaToNodeWithoutLock`